MyApplication/app/src/main/java/com/jonathanlieblich/myapplication/MainActivity.java

```diff
@@ -0,0 +1,108 @@
+    private Button mCereal, mChocolate, mTea;
+    private RecyclerView mRecycler;
+    private ProductViewAdapter mAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mCereal = (Button)findViewById(R.id.cereal);
+        mChocolate= (Button)findViewById(R.id.chocolate);
+        mTea = (Button)findViewById(R.id.tea);
+
+        View.OnClickListener listener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                switch(view.getId()) {
+                    case R.id.cereal:
+                        getProductRequest(CEREAL);
+                        break;
+                    case R.id.chocolate:
+                        getProductRequest(CHOCOLATE);
+                        break;
+                    case R.id.tea:
+                        getProductRequest(TEA);
+                        break;
+                }
+            }
+        };
+        mCereal.setOnClickListener(listener);
+        mChocolate.setOnClickListener(listener);
+        mTea.setOnClickListener(listener);
+    }
+
+    private void getProductRequest(String s) {
+        OkHttpClient client = new OkHttpClient();
+
+        Request request = new Request.Builder().url(URL_START+s).build();
+
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Call call, IOException e) {
+                e.printStackTrace();
+            }
+
+            @Override
+            public void onResponse(Call call, Response response) throws IOException {
+                if(!response.isSuccessful()) {
+                    throw new IOException("Unexpected code: "+response);
+                }
+
+                try {
+                    List<String> items1 = new ArrayList<>();
+                    JSONObject prim = new JSONObject(response.body().string());
+                    JSONArray products = prim.getJSONArray("items");
+                    for(int i=0;i<products.length();i++) {
+                        JSONObject item = products.getJSONObject(i);
+                        String itemName = item.getString("name");
+                        items1.add(itemName);
+                    }
+                    final List<String> items = items1;
+                    MainActivity.this.runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            mRecycler = (RecyclerView)findViewById(R.id.recycler_view);
+                            LinearLayoutManager layout = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL, false);
+                            mAdapter = new ProductViewAdapter(items);
+                            mRecycler.setLayoutManager(layout);
+                            mRecycler.setAdapter(mAdapter);
```

Comment:
Don't create a new adapter every time the data is updated; this is very inefficient. Instead, set the layout manager and adapter just once in `onCreate()` then here you can just update the data list and call `notifyDataSetChanged()` on the adapter.